packages/bff/src/services/selfcareService.ts

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+
+import { UserId } from "pagopa-interop-models";
+import {
+  InstitutionResource,
+  ProductResource,
+  SelfcareV2Client,
+  UserResource,
+} from "pagopa-interop-selfcare-v2-client";
+import { userNotFound } from "../model/domain/errors.js";
+
+export function selfcareServiceBuilder(selfcareV2Client: SelfcareV2Client) {
+  return {
+    async getSelfcareUser(
+      userId: UserId,
+      userIdQuery: string,
+      institutionId: string
+    ): Promise<UserResource> {
+      const users = await selfcareV2Client.getInstitutionProductUsersUsingGET({
+        params: { institutionId },
+        queries: {
+          userIdForAuth: userId,
+          userId: userIdQuery,
+        },
+      });
+
+      const user = users.at(0);
```

Comment:
```suggestion
      const user = users[0];
```
I would just access the array like this since it's shorter and the `.at` doesn't add any value

Reply:
Thanks for the suggestion, however i believe that using `.at` in this context actually provides some additional safety benefits that we should consider: in TypeScript using `.at` gives you `T | undefined` instead of just `T` (which may actually be undefined) so it is technically safer because the inferred type forces you to check if the value returned exists.

Reply:
> Thanks for the suggestion, however i believe that using `.at` in this context actually provides some additional safety benefits that we should consider: in TypeScript using `.at` gives you `T | undefined` instead of just `T` (which may actually be undefined) so it is technically safer because the inferred type forces you to check if the returned value exists.

You're right, thanks for letting me know! Then let's leave it like it is

Reply:
This made me notice that we don't have the `noUncheckedIndexedAccess` typescript rule active for this project. Maybe we should change that 😁